modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/CatalogSqlSchemaManager.java

```diff
@@ -140,12 +143,22 @@ private static SchemaPlus createSqlSchema(int version, CatalogSchemaDescriptor s
 
             IgniteTable schemaTable = new IgniteTableImpl(tableName, tableId, version, descriptor, statistic, tableIndexMap);
 
-            schemaTables.add(schemaTable);
+            schemaDataSources.add(schemaTable);
+        }
+
+        for (CatalogSystemViewDescriptor systemViewDescriptor : schemaDescriptor.systemViews()) {
+            int viewId = systemViewDescriptor.id();
+            String viewName = systemViewDescriptor.name();
+            TableDescriptor descriptor = createTableDescriptorForSystemView(systemViewDescriptor);
+
+            IgniteSystemView schemaTable = new IgniteSystemViewImpl(viewName, viewId, version, descriptor, Statistics.UNKNOWN);
```

Comment:
Is seems we could have special statistics realization for system views. At least we know distribution

Reply:
Unfortunately, there is no statistics for system views, because the API asks user to provide it a function that returns data (see SystemView's dataProvider property).

Reply:
yes, but statistics it not only about concrete rows, it also knowns about data structure and distribution, both of them don't depends on dataprovider

Reply:
Thanks. Fixed.